docs/development.md

```diff
@@ -200,6 +201,118 @@ use (
 )
 ```
 
+## Providers development best practices
+
+The more time we spent building providers, the more learn about how to do that better in the future. Here we described learnings that would help you get started with providers development.
+
+### Referencing MQL resources
+Often we have a top-level MQL resource, which we want to reference in another top-level resource. 
+
+For example, GCP networks can be retrieved for a project. That is a top-level resource:
+```
+// GCP Compute Engine
+private gcp.project.computeService {
+   // Google Compute Engine VPC network in a project
+   networks() []gcp.project.computeService.network
+}
+```
+
+However, we have a reference to a GCP network in a GCP Compute Address. This allows us to quickly navigate to the network in which an address is created:
+```
+private gcp.project.computeService.address {
+  // Static IP address
+  address string
+
+  // Network in which to reserve the address
+  network() gcp.project.computeService.network
+}
+```
+
+The simple wait to implement the reference would be to call the GCP API every time `gcp.project.computeService.address.network` is executed. However, this means we are going to be doing an excessive amount of API calls when scanning large GCP projects. If we have 10 addresses, this would mean we will do 10 separate API calls to get the network for each of them.
```

Comment:
```suggestion
The simple way to implement the reference would be to call the GCP API every time `gcp.project.computeService.address.network` is executed. However, this would generate an excessive amount of API calls when scanning large GCP projects. If we have 10 addresses, this would mean 10 separate API calls to get the network, one for each of them.
```